### DISCUSSION TITLE
Idea about PMU enablement on Armv8-A CPUs using software

### DISCUSSION THREAD

**Discussion by Jiaming97** [ArmDeveloperEcosystem/arm-learning-paths#474] [Ideas for new Learning Paths]
Hi all, 
I am an Arm employee. I have written a tutorial about PMU enablement on Armv8-A CPUs using software.  It covers basic workflow of PMU usage, three typical examples(to enable in u-boot/TF-A/Linux) on the real Armv8-A platform. I think it a good chance to post it on the learn.arm.com, since my tutorial is a step-by-step one with software enablement. Do you guys think it a good idea?

Thanks,
Jiaming Guo

**Comment by jasonrandrews** [comment]
Hi @Jiaming97, 
Thank you for asking about Learning Paths. Your topic sounds great. Not sure if you will cover Perf, but if so we recently added an [install guide](https://learn.arm.com/install-guides/perf/) for Perf and you can reference that if Perf is needed for your tutorials. My only other suggestion is to try to use hardware that is available. 

I'm happy to review anything along the way or give you a tutorial on how to get setup. Your best resources are the [YouTube video](https://youtu.be/uKki6N96Sqw?si=_XYdYj9eop9IAZM-) and the [Learning Path about how to create Learning Paths](https://learn.arm.com/learning-paths/cross-platform/_example-learning-path/).

Looking forward to your contribution,
Jason

**Comment by Jiaming97** [comment]
Hi @jasonrandrews,
Thank you for providing meaningful information for me. 

Actually my tutorial on PMU enablement is not about how to use tool (e.g. perf) but how to program it. I noticed that even though we provide example code snippets, customers still ask similar questions on failure of PMU enablement. One of the reason is that the software stack is not considered in these code snippets. 

So I want to provide a tutorial on how to enable PMU with a real Armv8-A platform as an example. I will give examples on how to program to enable PMU in TF-A/u-boot/Linux, then give the test result on a real platform. Because I have already written the tutorial, I will setup and submit contribution soon.

Thanks
Jiaming Guo

**Reply by jasonrandrews** [reply]
Hi @Jiaming97 
The content sounds excellent, looking forward to seeing it. 
Jason
